improve description of API review process

lawli3t opened the pull request:
<!--
Check relevant points but **please do not remove entries**.
-->

## Basics

<!--
These points need to be fulfilled for every PR.
-->

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

<!--
If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you,
but we cannot accept PRs that do not fulfill the basics.
-->

## Checklist

<!--
For docu fixes, spell checking, and similar none of these points below
need to be checked.
-->

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

<!--
Reviewers should check the following.
-->

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

<!--
If you are already Elektra developer, please adjust the labels.
Otherwise, write a comment and it will be done for you.
-->

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

lawli3t committed:
improve description of API review process

markus2330 commented on doc/contrib/api/reviews/README.md:
```diff
@@ -7,10 +7,19 @@ Below you can find a short description of the review process.
 This review is performed for each function separately.
 Each function should be evaluated according to the checklist.
+The template contains a short section with metadata about the review (time, reviewer) that must also be filled out.
+A function can be reviewed multiple times by different people in order to improve upon the results. Different people find different issues and more reviewers usually means that more problems get uncovered.
```
document guidelines want newlines for every sentence:

```suggestion
A function can be reviewed multiple times by different people in order to improve upon the results.
Different people find different issues and more reviewers usually means that more problems get uncovered.
```

markus2330 commented on doc/contrib/api/reviews/README.md:
```diff
@@ -7,10 +7,19 @@ Below you can find a short description of the review process.
+Multiple reviews of the same function should be put into the same file below each other.
+The review should preferrably be performed by a person that did not directly work on the design of the function that is being reviewed.
+If possible, the review should be conducted by at least two people, where one person assumes the role of the moderator and one person assumes the role of the reviewer.
+The moderator would preferably be a person that took part in the design of the function. This enables the reviewer to as for clarification on design decisions that seem unclear.
```
```suggestion
The moderator would preferably be a person that took part in the design of the function.
This enables the reviewer to as for clarification on design decisions that seem unclear.
```

markus2330 commented on scripts/api_review/README.md:
```diff
@@ -1,3 +1,9 @@
 This python script generates new review files from the given template.
+
+The script takes a list of methods as well as a template as input parameters.
+The list of methods should be a file where each line contains a signature of a function that should be reviewed. A review file will be generated for each line.
```
```suggestion
The list of methods should be a file where each line contains a signature of a function that should be reviewed.
A review file will be generated for each line.
```

markus2330 commented on scripts/api_review/README.md:
```diff
@@ -1,3 +1,9 @@
+
+The script takes a list of methods as well as a template as input parameters.
+The list of methods should be a file where each line contains a signature of a function that should be reviewed. A review file will be generated for each line.
+The template gets passed two variables: signature and name of the function which can be accessed in the template via `{signature}` and `{name}` respectively.
```
```suggestion
The template gets passed two variables:
signature and name of the function which can be accessed in the template via `{signature}` and `{name}` respectively.
```

kodebach replied:
Technically, this is one sentence, even if there is a colon, since the part after the colon is not a full sentence (has no verb/predicate). You could also write it like this to make it more clear that it's just one sentence:

```suggestion
The template gets passed two variables, the signature and name of the function, which can be accessed in the template via `{signature}` and `{name}` respectively.
```

markus2330 reviewed (approved):
LGTM

markus2330 commented:
Thanks for the write-up! Could you please add a line for release notes?

kodebach reviewed (commented):
LGTM, but I have two suggestions:

1. Some more line breaks to form paragraphs in the `README.md`. Currently the rendered version is just on huge wall of text.
2. The current review process is clearly aimed at an in-person or video conference setting. That makes sense for a one-time review (like for a thesis), but for on going reviews and reviews of new/proposed APIs it could be annoying. I think the process could easily be adapted to work in a PR setting.:
   
   The "moderator" is simply the person that proposed an API. Each reviewer opens a new PR which adds to the API review file (*). In the PR they can ask the "moderator" questions and the "moderator" can give additional clarifications, before a review is merged.
   
   The process would probably work a bit worse and more changes might be needed in future (e.g. for proposed APIs use the review template as guidelines for a normal code review on Github). But at least we'd have something that doesn't need scheduling of synchronous meetings.

    (*) if we use a single file for multiple reviews, we may run into merge conflict issues again.

lawli3t commented:
> 2\. The "moderator" is simply the person that proposed an API. Each reviewer opens a new PR which adds to the API review file (*). In the PR they can ask the "moderator" questions and the "moderator" can give additional clarifications, before a review is merged.

Sounds like a good idea to me and probably causes less friction during the review process.

I've changed the README to reflect this way of reviewing and added your proposed changes. I've also structured everything a bit better and improved the wording of some sentences.

kodebach reviewed (approved):
LGTM

lawli3t committed:
add release notes
